Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs

```diff
@@ -310,139 +356,147 @@ public void TryClose(IEntity user)
-                    return true;
+                return false;
             }
 
-            return false;
+            return SafetyCheck();
         }
 
-        public bool Close()
+        /// <summary>
+        /// Checks if we are allowed to crush people, and if something is colliding with the door.
+        /// </summary>
+        /// <returns>True if we shouldn't worry about something colliding with us, false if we should.</returns>
+        public bool SafetyCheck()
         {
-            bool shouldCheckCrush = false;
-            if (Owner.TryGetComponent(out IPhysicsComponent? physics))
-                physics.CanCollide = true;
-
-            if (_canCrush && physics != null &&
-                physics.IsColliding(Vector2.Zero, false))
+            if (Safety && Owner.TryGetComponent(out IPhysicsComponent? physics))
             {
-                if (Safety)
+                // WE ARE NOT ACTUALLY BECOMING COLLIDEABLE. WE JUST NEED TO SEE IF WE SHOULD DELAY CLOSING DUE TO A PERSON.
+                var storedCanCollide = physics.CanCollide;
+                physics.CanCollide = true;
+                if (physics.IsColliding(Vector2.Zero, false))
                 {
-                    physics.CanCollide = false;
+                    physics.CanCollide = storedCanCollide;
                     return false;
                 }
-
-                // check if we crush someone while closing
-                shouldCheckCrush = true;
+                physics.CanCollide = storedCanCollide;
             }
+            return true;
+        }
 
+        /// <summary>
+        /// Closes the door. Does not check if this is possible.
+        /// </summary>
+        public void Close()
+        {
             State = DoorState.Closing;
-            OpenTimeCounter = 0;
-            SetAppearance(DoorVisualState.Closing);
-            if (_occludes && Owner.TryGetComponent(out OccluderComponent? occluder))
-            {
-                occluder.Enabled = true;
-            }
+            _openTimeCounter = 0;
 
             _cancellationTokenSource?.Cancel();
             _cancellationTokenSource = new();
-
             Owner.SpawnTimer(CloseTimeOne, async () =>
             {
-                if (shouldCheckCrush && _canCrush)
-                {
-                    CheckCrush();
-                }
-
-                if (Owner.TryGetComponent(out AirtightComponent? airtight))
+                // if somebody walked into the door as it was closing, and we don't crush things
+                if (!SafetyCheck())
                 {
-                    airtight.AirBlocked = true;
+                    Open();
+                    return;
                 }
 
-                if (Owner.TryGetComponent(out IPhysicsComponent? body))
+                OnPartialClose();
+                await Timer.Delay(CloseTimeTwo, _cancellationTokenSource.Token);
+                if (Occludes && Owner.TryGetComponent(out OccluderComponent? occluder))
```

Comment:
componentdependencies

Reply:
Same as above, I don't think the directional doors necessarily occlude right?